app/presenters/idv/by_mail/letter_enqueued_presenter.rb

```diff
@@ -0,0 +1,48 @@
+module Idv
+  module ByMail
+    class LetterEnqueuedPresenter
+      include ActionView::Helpers::TranslationHelper
+      include Rails.application.routes.url_helpers
+
+      def initialize(idv_session)
+        @pii = idv_session.pii_from_doc ||
+               idv_session.user_session.dig('idv/in_person', :pii_from_user) ||
+               Pii::Cacher.new(idv_session.current_user, idv_session.user_session).
+                 fetch(idv_session.current_user&.gpo_verification_pending_profile&.id)
+        @sp = idv_session.service_provider
+      end
+
+      def address_lines
+        [
+          pii[:address1],
+          pii[:address2],
+          "#{pii[:city]}, #{pii[:state]} #{pii[:zipcode]}",
+        ].compact
+      end
+
+      def button_text
+        if sp
+          t('idv.cancel.actions.exit', app_name: APP_NAME)
+        else
+          t('idv.buttons.continue_plain')
+        end
+      end
+
+      def button_destination
+        if sp
+          return_to_sp_cancel_path(step: :get_a_letter, location: :come_back_later)
+        else
+          account_path
+        end
+      end
+
+      def url_options
```

Comment:
I think it's preferable to pass in `url_options` from the controller as a constructor argument. I'm not sure if this would preserve locale in generated URLs, for example.

Reply:
Changed to pass in to the constructor.